torch/csrc/distributed/c10d/comm.h

```diff
@@ -30,7 +30,11 @@ class GradBucket {
 // DDP's c10d reducer allows communcation hooks defined as a sub class
 // of CommHookInterface. CommHookInterface is an abstract class and can
 // be used to implement both Python and CPP hooks.
+#ifdef _WIN32
+struct CommHookInterface {
```

Comment:
any reason we cannot expose DDP hook interface from libtorch when using Windows?

Reply:
By default build on windows, C10_BUILD_SHARED_LIBS is defined and CAFFE2_BUILD_MAIN_LIB is not defined. So TORCH_API is defined as **__declspec(dllimport)** which will lead a compile error. Should I add a compile option for CAFFE2_BUILD_MAIN_LIB in caffe2/CMakeLists.txt?

Reply:
> Should I add a compile option for CAFFE2_BUILD_MAIN_LIB in caffe2/CMakeLists.txt?

Sorry, I don't know the answer. @malfet do you know what's the preferred way here?

Reply:
Sounds like `C10_EXPORT` macro would solve everyones problems here, wouldn't it? (Although in order to support DDP hooks on Windows we perhaps need to add something like TORCH_PYTHON_API)